### PR TITLE
Fix Snowpark type casting

### PR DESCRIPTION
**Pull request by sfc-gh-turbaszek**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [x] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually on MacOS.
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually on Windows.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [ ] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Adjusts the test to work consistently on CI and QA environments. Type conversion for snowpark was reversed: user input -> snowflake type.

### CODE REVIEW

**Comment by sfc-gh-turbaszek** [comment]
Skipping the one of NA test because is failing also on other PRs
https://github.com/snowflakedb/snowflake-cli/pull/1691

**Review by sfc-gh-jsikorski — APPROVED** [review]

**Review by sfc-gh-astus — APPROVED** [review]

**Review by sfc-gh-jsikorski — APPROVED** [review]

File: tests_integration/nativeapp/test_large_upload.py [hunk 1]
```diff
@@ -45,6 +42,10 @@ def test_large_upload_skips_reupload(
     Ensure that files uploaded in multiple parts are not re-uploaded unnecessarily.
     This test will currently fail when run on a non-AWS deployment.
     """
+    from snowflake.cli._plugins.nativeapp.v2_conversions.compat import (
+        _pdf_v2_to_v1,
+    )
```
Comment (sfc-gh-turbaszek) [inline tests_integration/nativeapp/test_large_upload.py:47]: This was causing import errors

File: tests_integration/nativeapp/test_bundle.py [hunk 2]
```diff
@@ -271,6 +273,7 @@ def test_nativeapp_bundle_throws_error_on_too_many_files_to_dest(template_setup)
 
 # Tests handling of no artifacts
 @pytest.mark.integration
+@pytest.mark.skip("Flaky test, needs to be fixed")
```
Comment (sfc-gh-turbaszek) [inline tests_integration/nativeapp/test_bundle.py:276]: This test was consistently failing

**Review by sfc-gh-bdufour — APPROVED** [review]
Approving the test fixes on behalf of NADE

**Pull request merged**